apicast/src/provider.lua

```diff
@@ -385,8 +385,15 @@ function _M.access(service)
     ngx.var.cached_key = concat({service.id, params.app_id, params.app_key}, ':')
 
   elseif backend_version == 'oauth' then
-    ngx.var.access_token = parameters.access_token
-    params.access_token = parameters.access_token
+    if credentials.location == 'headers' then
```

Comment:
Maybe we can just use Authorization header to be used every time, not only when headers are used. Also for the other authorization methods. 
That could make it consistent across auth methods.  And later introduce setting that enforces passing the Authorization header.